rcl/test/rcl/test_publisher.cpp

```diff
@@ -490,3 +535,291 @@ TEST_F(CLASSNAME(TestPublisherFixture, RMW_IMPLEMENTATION), test_invalid_publish
     rcl_publish_serialized_message(nullptr, &serialized_msg, nullptr));
   rcl_reset_error();
 }
+
+// Mocking rmw_publisher_count_matched_subscriptions to make
+// rcl_publisher_get_subscription_count fail
+TEST_F(
+  CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION),
+  test_mock_publisher_get_subscription_count)
+{
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rcl", rmw_publisher_count_matched_subscriptions, RMW_RET_BAD_ALLOC);
+
+  // Now normal usage of the function rcl_publisher_get_subscription_count returning
+  // unexpected RMW_RET_BAD_ALLOC
+  size_t count_size;
+  EXPECT_EQ(
+    RCL_RET_BAD_ALLOC, rcl_publisher_get_subscription_count(&publisher, &count_size));
+  rcl_reset_error();
+}
+
+// Mocking rmw_publisher_assert_liveliness to make
+// rcl_publisher_assert_liveliness fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_assert_liveliness) {
+  auto mock = mocking_utils::patch_and_return(
+    "lib:rcl", rmw_publisher_assert_liveliness, RMW_RET_ERROR);
+
+  // Now normal usage of the function rcl_publisher_assert_liveliness returning
+  // unexpected RMW_RET_ERROR
+  EXPECT_EQ(
+    RCL_RET_ERROR, rcl_publisher_assert_liveliness(&publisher));
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+}
+
+// Mocking rmw_publish to make rcl_publish fail
+TEST_F(CLASSNAME(TestPublisherFixtureInit, RMW_IMPLEMENTATION), test_mock_publish) {
+  auto mock = mocking_utils::patch_and_return("lib:rcl", rmw_publish, RMW_RET_ERROR);
+
+  // Test normal usage of the function rcl_publish returning unexpected RMW_RET_ERROR
+  test_msgs__msg__BasicTypes msg;
+  test_msgs__msg__BasicTypes__init(&msg);
+  msg.int64_value = 42;
+  ret = rcl_publish(&publisher, &msg, nullptr);
+  test_msgs__msg__BasicTypes__fini(&msg);
+  EXPECT_EQ(RCL_RET_ERROR, ret) << rcl_get_error_string().str;
+  EXPECT_TRUE(rcl_error_is_set());
+  rcl_reset_error();
+}
+
+// Mocking rmw_publish_serialized_message to make rcl_publish_serialized_message fail
```

Comment:
Several of these unit tests have long stretches of code without white space or comments, could you organize things a bit more so these files are easier to parse?

Reply:
See 98defc1, mainly added some new lines to separate `init` of the variables vs specific test code